pytest_motor/plugin.py

```diff
@@ -50,32 +50,13 @@ async def root_directory(pytestconfig: PytestConfig) -> Path:
 
 
 @pytest.fixture(scope='session')
-async def mongod_binary(root_directory: Path) -> Path:  # pylint: disable=redefined-outer-name
+async def mongod_binary(root_directory: Path) -> Path:
     """Return a path to a mongod binary."""
-    destination: Path = root_directory.joinpath('.mongod')
-    os_ver, mongo_ver = _mongo_ver()
-    mongod_binary_relative_path = f'{mongo_ver}/bin/mongod'
-    mongod_binary_path = destination.joinpath(mongod_binary_relative_path)
-
-    if not mongod_binary_path.exists():
-        download_url = f'https://fastdl.mongodb.org/{os_ver}/{mongo_ver}.tgz'
-        async with aiohttp.ClientSession() as session:
-            async with session.get(download_url) as resp:
-                with tempfile.TemporaryFile(mode='w+b') as binary_file:
-                    # Read by chunks to avoid big RAM consumption
-                    while True:
-                        # read by 100 bytes
-                        chunk = await resp.content.read(100)
-                        if not chunk:
-                            break
-                        binary_file.write(chunk)
-                    binary_file.flush()
-                    binary_file.seek(0)
-                    # Extract tar
-                    with tarfile.open(fileobj=binary_file) as tar:
-                        tar.extract(member=mongod_binary_relative_path, path=destination)
-
-    return mongod_binary_path
+    destination: Path = root_directory / '.mongod'
+    binary = MongodBinary(destination=destination)
+    if not binary.path.exists():
+        await binary.download_and_unpack()
+    return binary.path
```

Comment:
I like how much this has been simplified. One small suggestion would be to add an `exists` property to `MongodBinary`. It would read slightly cleaner: `if not binary.exists:`